p2p/peer.py

```diff
@@ -342,7 +343,81 @@ def is_closing(self) -> bool:
     async def _cleanup(self) -> None:
         self.close()
 
+    async def boot(self) -> None:
```

Comment:
I think we should wait for the `started` event or for `self.is_operational` here, no?

Reply:
Since `boot` is called from within the `_run`, `started` would always be set here.  That could change if code gets refactored, so I added a check to see if `started` was already set.